cdap-data-fabric/src/main/java/co/cask/cdap/data2/transaction/stream/FileStreamAdmin.java

```diff
@@ -204,14 +204,26 @@ public void upgrade() throws Exception {
 
   @Override
   public StreamConfig getConfig(Id.Stream streamId) throws IOException {
-    Location streamLocation = getStreamBaseLocation(streamId);
-    Preconditions.checkArgument(streamLocation.isDirectory(), "Stream '%s' does not exist.", streamId);
-    return loadConfig(streamLocation);
+    Location configLocation = getConfigLocation(streamId);
+    Preconditions.checkArgument(configLocation.exists(), "Stream '%s' does not exist.", streamId);
+
+    StreamConfig config = GSON.fromJson(
+      CharStreams.toString(CharStreams.newReaderSupplier(Locations.newInputSupplier(configLocation), Charsets.UTF_8)),
+      StreamConfig.class);
+
+    int threshold = config.getNotificationThresholdMB();
+    if (threshold <= 0) {
```

Comment:
It's existing code. It is the default value of long field when deserializing using Gson and yet the field is missing.